ambari-server/src/main/java/org/apache/ambari/server/controller/internal/RequestResourceProvider.java

```diff
@@ -324,6 +340,22 @@ public RequestStatusResponse invoke() throws AmbariException, AuthorizationExcep
     return resources;
   }
 
+  private void authorizeGetResources() throws SystemException {
+    final Set<RoleAuthorization> requiredAuthorizations = Sets.newHashSet();
+    for (String allowedPermissionName : ALLOWED_PERMISSION_NAMES) {
+      PermissionEntity permission = permissionDao.findByName(allowedPermissionName);
+      if (permission != null) {
+        for (RoleAuthorizationEntity roleAuthorization : permission.getAuthorizations()) {
+          requiredAuthorizations.add(RoleAuthorization.translate(roleAuthorization.getAuthorizationId()));
+        }
+      }
+    }
+
+    if (!AuthorizationHelper.isAuthorized(null, null, requiredAuthorizations)) {
```

Comment:
Rather than testing authorization using roles, we test authorization using privileges (or authorizations - RoleAuthorization).  Privileges are assigned to roles and therefore new roles can be added without needing to update the code.

There are two cases here... one for Ambari-level requests and one for cluster-level requests.   Ambari-level requests have no resource associated with them, where as cluster-level requests have a cluster resource associated with them. 

For the **Ambari-level** case, a new authorization should be created... AMBARI.VIEW_REQUEST_DETAILS (or something like that). This privilege should be added to the AMBARI.ADMINISTRATOR role. 

Once this is done, the authorization check would look something like:

```
AuthorizationHelper.isAuthorized(ResourceType.AMBARI, null, RoleAuthorization.AMBARI_VIEW_REQUEST_DETAILS)
```

For the **cluster-level** case, a new set of authorizations should be created... CLUSTER.VIEW_REQUEST_DETAILS, SERVICE.VIEW_REQUEST_DETAILS, and HOST.VIEW_REQUEST_DETAILS (or something like that). This privileges should be added to the following roles: 
- AMBARI.ADMINISTRATOR
- CLUSTER.ADMINISTRATOR
- CLUSTER.OPERATOR
- SERVICE.ADMINISTRATOR
- SERVICE.OPERATOR
- CLUSTER.USER

Once this is done, the authorization check would look something like:

```
AuthorizationHelper.isAuthorized(ResourceType.CLUSTER, <cluster resource id>, Enum.setOf(RoleAuthorization.CLUSTER_VIEW_REQUEST_DETAILS, RoleAuthorization.SERVICE_VIEW_REQUEST_DETAILS, and RoleAuthorization.HOST_VIEW_REQUEST_DETAILS))
```

Reply:
Got it; I'll update the code and submit a new PR as soon as I can

Reply:
We have some mechanism in AbstractAuthorizedResourceProvider (and this class extends it) to check if get/create/delete/etc is allowed. The setRequiredGetAuthorizations() can be used to set the required role authorizations to access the resource. Can't we use that mechanism instead of checking this manually?

Reply:
@rlevas I see, thanks for the explanation.